crates/llvm-context/src/polkavm/context/function/runtime/immutable_data_load.rs

```diff
@@ -34,9 +36,17 @@ where
     }
 
     fn into_llvm(self, context: &mut Context<D>) -> anyhow::Result<()> {
-        context.set_current_function(runtime::FUNCTION_LOAD_IMMUTABLE_DATA)?;
+        context.set_current_function(runtime::FUNCTION_LOAD_IMMUTABLE_DATA, None)?;
         context.set_basic_block(context.current_function().borrow().entry_block());
 
+        if context.debug_info().is_some() {
+            context.builder().unset_current_debug_location();
+            let func_scope = context
+                .set_current_function_debug_info(runtime::FUNCTION_LOAD_IMMUTABLE_DATA, 0)?
+                .as_debug_info_scope();
+            context.push_debug_scope(func_scope);
+        }
+
```

Comment:
This code is duplicated 3x. Since it is needed every time we lower functions it should be a helper on context instead.